File: RepoTransBench/target_projects/Python/C++/rust-lang_homu/tests/original/test_main_test.cpp
#include <gtest/gtest.h>
#include <string>
#include <algorithm>
// #include "main.h"

// Placeholder for process_input, pr_body_contains until src logic exists
std::string process_input(const std::string& s) {
    std::string rev = s;
    std::reverse(rev.begin(), rev.end());
    // If palindrome, return as is
    if (rev == s) return s;
    return rev;
}
bool pr_body_contains(const std::string& body, const std::string& key) {
    return body.find(key) != std::string::npos;
}

TEST(MainTest, ProcessInputReverseExisting) {
    EXPECT_EQ(process_input("test"), "tset");
}
TEST(MainTest, ProcessInputPalindromeExisting) {
    EXPECT_EQ(process_input("abba"), "abba");
}

File: RepoTransBench/target_projects/Python/Go/rust-lang_homu/tests/original/test_main_test.go
package original

import (
	"testing"
)

// Simulated processInput reverses unless it's a palindrome.
func processInput(s string) string {
	// Check if palindrome
	n := len(s)
	isPalindrome := true
	for i := 0; i < n/2; i++ {
		if s[i] != s[n-1-i] {
			isPalindrome = false
			break
		}
	}
	if isPalindrome {
		return s
	}
	// Reverse
	runes := []rune(s)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}

func TestProcessInputReverseExisting(t *testing.T) {
	result := processInput("test")
	if result != "tset" {
		t.Errorf("processInput(\"test\") = %q, want %q", result, "tset")
	}
}

func TestProcessInputPalindromeExisting(t *testing.T) {
	result := processInput("abba")
	if result != "abba" {
		t.Errorf("processInput(\"abba\") = %q, want %q", result, "abba")
	}
}